compiler/qsc_frontend/src/resolve.rs

```diff
@@ -569,6 +582,68 @@ impl Resolver {
                 scope.tys.insert(Rc::clone(&name.name), id);
                 scope.terms.insert(Rc::clone(&name.name), id);
             }
+            ast::ItemKind::Export(export) => {
+                // resolve the exported item and insert the vec ident into the names table, so we can access it in
+                // lowering
+                for item in export.items() {
+                    let resolved_item = match resolve(
+                        NameKind::Term,
+                        &self.globals,
+                        self.locals.get_scopes(&self.curr_scope_chain),
+                        &item.name,
+                        &item.namespace,
+                    ) {
+                        Ok(res) => res,
+                        Err(_) => {
+                            // try to see if it is a type
+                            match resolve(
+                                NameKind::Ty,
+                                &self.globals,
+                                self.locals.get_scopes(&self.curr_scope_chain),
+                                &item.name,
+                                &item.namespace,
+                            ) {
+                                Ok(res) => res,
+                                Err(err) => {
+                                    self.errors.push(err);
+                                    continue;
+                                }
+                            }
+                        }
+                    };
+
+                    let scope = self.current_scope_mut();
+
+                    let resolved_item_id = match resolved_item {
+                        Res::Item(
+                            ItemId {
+                                package: Some(_), ..
+                            },
+                            _,
+                        ) => {
+                            self.errors.push(Error::ExportedExternalItem(item.span));
+                            continue;
+                        }
+                        Res::Item(id, _) => id,
+                        _ => {
+                            self.errors.push(Error::ExportedNonItem(item.span));
+                            continue;
+                        }
+                    };
+
+                    scope
+                        .terms
+                        .insert(Rc::clone(&item.name.name), resolved_item_id);
```

Comment:
Why is this needed? The terms table should have been updated when the item was bound in the first place.

Reply:
Looks like this makes it so that if you're exporting an item from a _different_ namespace, the name is made available in the current namespace.

Not sure if this is intentional behavior, but it at least needs to be tweaked a little, e.g. to avoid the ambiguous resolution issue from my other comment.

There should probably also be a unit test for it, so that _something_ fails when I comment out this line